### PR TITLE
Feature/song detail

### PR DESCRIPTION
**Pull request by miggelucas**
Hi folks. Sorry for delaying the delivery of this PR. This PR Isn't pack with all features I thought for this view. In fact, the more I developed and implemented features, the more I realized that would need to add some more (like if the user can edit a song info, he also should be able to manage the pedals. 
There are some missing points, and if you look around some bugs should appear. The view and viewModels are very tightly coupled, which leads to some unexpected behaviors. I believe that once we have a service layer responsible SOT of user Data we could have better control of the app. Never the less, please, feel free to suggest any modification or points of improvement in the code.

Simulator Screen Recording - iPhone 12 16 1 - 2023-11-29 at 11 39 06

## What I Have Done
* SelectPedalView rework
* KnobView and Grid rework
* SongDetailView displays song info
* SongDetailView displays pedalsKnobs
* SongDetailView edit song and pedals feature
* SongDetailView edit pedal knobs feature
* Modifications of Song can be saved to Song SOT
* SongDetailView presents alert when saving song fails
* SongDetailView animations
* SongDetailViewModel responsable for dealing with SongDetailViewLogic
* SongDetailViewModel tests

### CODE REVIEW

File: PedalCore/Views/Songs/SongDetail/SongDetailView.swift [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+                            KnobsGridView(knobs: $pedal.knobs, knobStyle: viewModel.isInEditingKnobsMode ? .editing : .reference)
+                                .allowsHitTesting(viewModel.isInEditingKnobsMode)
+                        }
+                    }
+                    .frame(maxWidth: .infinity,
+                           alignment: .leading)
+                    .padding()
+                }
+            }
+            .background(
```
Comment (matheusmigge) [inline PedalCore/Views/Songs/SongDetail/SongDetailView.swift]:
```suggestion
            }
            .padding(.bottom, 20)
            .background(
```
I would like to suggest a bottom padding, just to for an aesthetic reason!

**Review by matheusmigge — APPROVED** [review]
Your PR is pretty good, @miggelucas!! You made an AMAZING job! Congratulations! Just made a single aesthetic suggestion, just check if you agree

File: PedalCore/Views/Songs/SongDetail/SongDetailView.swift [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+                Rectangle()
+                    .cornerRadius(10)
+                    .foregroundStyle(colorScheme == .light ?  Color.white : Color.black)
+                    .shadow(color: .accentColor, radius: viewModel.isInEditing ? 5 : 0)
```
Comment (matheusmigge) [inline PedalCore/Views/Songs/SongDetail/SongDetailView.swift]:
```suggestion
                    .shadow(color: viewModel.isInEditing ? .accentColor : Color.clear, radius: viewModel.isInEditing ? 5 : 0)
```

File: PedalCore/Views/Songs/SongDetail/SongDetailView.swift [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+                    .foregroundStyle(.primary)
+                    .padding(5)
+                
+            }
+            .allowsHitTesting(viewModel.isInEditingSongMode)
+            .background(
+                Rectangle()
+                    .cornerRadius(10)
+                    .foregroundStyle(colorScheme == .light ? Color.white : Color.black)
+                    .shadow(color: .accentColor, radius: viewModel.isInEditingSongMode ? 5 : 0)
```
Comment (matheusmigge) [inline PedalCore/Views/Songs/SongDetail/SongDetailView.swift]:
```suggestion
                    .shadow(color: viewModel.isInEditingSongMode ? .accentColor: Color.clear, radius: viewModel.isInEditingSongMode ? 5 : 0)
```

**Review by matheusmigge — APPROVED** [review]
Just made some other design suggestions

**Review by matheusberger — APPROVED** [review]
Amazing PR! It looks incredible, loved the animations. Please feel free address the stuff pointed out by @matheusmigge in this PR, or merge it as-is and create new cards in our trello to track those improvements.

**Comment by miggelucas** [comment]
Thanks for the feedback, guys! All suggestions were accepted

**Pull request merged**